docs/src/density_interface.md

```diff
@@ -0,0 +1,9 @@
+# Support for DensityInterface
+
+`Distributions` supports [`DensityInterface`](https://github.com/JuliaMath/DensityInterface.jl) for distributions: `DensityInterface.logdensityof(d, x)` is equivalent to `logpdf(d, x)` and `DensityInterface.densityof(d::Distribution, x)` is equivalent to `pdf(d, x)` for *single* variate values `x`.
+
+To get the log-density value at either a single variate value or at a set of variate values (implying a product distribution over `d`), use `logdensityof(IIDDensity(d), x)`, this is equivalent to `loglikelihood(d, x)`.
+
+```@docs
+IIDDensity
+```
```

Comment:
I don't think there's a real use case for `IIDDensity` yet apart from the potential use in Turing, so I think we should maybe not export it currently.

````suggestion
You can use
```julia
DensityInterface.logdensityof(Distributions.IIDDensity(d), x)
```
and
```julia
DensityInterface.densityof(Distributions.IIDDensity(d), x)
```
to evaluate the log-density and density of variate value(s) `x` that are assumed to be identically and independently
distributed according to distribution `d`. Here `x` can be either a single variate value or at a set of variate values.
`DensityInferface.logdensityof(Distributions.IIDDensity(d), x)` falls back to `loglikelihood(d, x)`.

```@docs
Distributions.IIDDensity
```
````